### PR TITLE
Fix type casting logic for read and write operations

### PR DESCRIPTION
**Pull request by raknay**
# Summery
PR to addresses [issue](https://github.com/DiceDB/dice/issues/367)
Fix type casting logic in read and write operations.

### CODE REVIEW

File: core/eval.go [hunk 1]
```diff
@@ -235,7 +235,19 @@ func evalMSET(args []string, store *Store) []byte {
...
+		// Cast the value properly based on the encoding type
+		var storedValue interface{}
+		switch oEnc {
+		case ObjEncodingInt:
+			storedValue, _ = strconv.ParseInt(value, 10, 64)
+		case ObjEncodingEmbStr, ObjEncodingRaw:
+			storedValue = value
+		default:
+			return Encode(fmt.Errorf("ERR unsupported encoding: %d", oEnc), false)
+		}
```
[the remark this answers is not in the document]
Comment (soumya-codes) [reply core/eval.go:248]: Great suggestion! @raknay, there is an existing function for byte arrays that abstracts the code into a function [here.](https://github.com/DiceDB/dice/blob/master/core/bitpos.go#L64). Please see if we can generalize this

**Comment by raknay** [comment]
@JyotinderSingh @soumya-codes I have gone through the review comments. Will push the changes by today.

**Comment by JyotinderSingh** [comment]
Hi @raknay, are you still actively working on this issue?

**Comment by AshwinKul28** [comment]
HI @raknay hope everything is now okay at your end. Can you please update this PR and also rebase with the master? Lets try to close it out ASAP, because a lot many changes are getting merged to master and it would be difficult for you to resolve merge conflicts every time. Thanks a lot for the contribution.

**Pull request closed without merging**